### PR TITLE
restructuring rfsm (lua) to be also used with rfsmGui

### PR DESCRIPTION
**Pull request by apaikan**
In this pull request:
 - the initialization of the iol (yarp.network, port opening, grammar, ..) has been moved to `IOL_Initialize()`  function 
- the `iol_main.lua` has been simplified to just lunch the rfsm 
- a new state for initializing iol has been added

### CODE REVIEW

**Comment by pattacini** [comment]
Hi @apaikan 

Would you be willing to adapt also the lua scripts in [`mobile`][1] subdirectory?

[1]: https://github.com/robotology/iol/tree/master/app/lua/mobile

File: app/scripts/iol.xml.template [hunk 1]
```diff
@@ -117,6 +119,18 @@
                 <port timeout="10">/iolStateMachineHandler/human:rpc</port>
                 <port timeout="10">/iolHelper/rpc</port>
             </dependencies>
+        </module> -->
+
+        <module>
+            <name>rfsmGui</name>
+            <node>console</node>
+            <parameters>-f <path to context>/iol/lua/iol_root_fsm.lua --run</parameters>
```
Comment (pattacini) [inline app/scripts/iol.xml.template]: that's bad... 😉 
we should implement the RF searching policy, somehow.

**Comment by apaikan** [comment]
@Ugo: fixed in commit d49595e !

**Comment by pattacini** [comment]
Hi @apaikan 
Is this PR fully tested?

**Pull request merged**